### PR TITLE
Add Untainted qualifier.

### PR DESCRIPTION
**Pull request by smillst**
Merge with https://github.com/typetools/checker-framework/pull/2883.

### CODE REVIEW

File: src/java.base/share/classes/java/lang/Object.java [hunk 1]
```diff
@@ -62,7 +63,7 @@
      * Constructs a new object.
      */
     @HotSpotIntrinsicCandidate
-    public Object() {}
+    public @Untainted Object() {}
```
Comment (wmdietl) [inline src/java.base/share/classes/java/lang/Object.java]: Should you also add tainting to `@AnnotatedFor`?

Comment (smillst) [reply src/java.base/share/classes/java/lang/Object.java]: No, all the methods have not been annotated, just this one.

Comment (wmdietl) [reply src/java.base/share/classes/java/lang/Object.java]: Our current suggestion is to fully annotate files: https://checkerframework.org/manual/#annotating-jdk
j.l.Object isn't very big, so wouldn't it be better to fully annotate it for the Tainting Checker?
The "Whenever you add a file" part no longer makes sense, as all possible files are already contained in the typetools/jdk repository.

Comment (smillst) [reply src/java.base/share/classes/java/lang/Object.java]: We will do that later.

Comment (wmdietl) [reply src/java.base/share/classes/java/lang/Object.java]: You should at least open an issue to keep track of this.

**Pull request merged**